native/explorer/src/series.rs

```diff
@@ -53,32 +53,38 @@ from_list!(s_from_list_str, String);
 macro_rules! from_list_float {
     ($name:ident, $type:ty, $module:ident) => {
         #[rustler::nif(schedule = "DirtyCpu")]
-        pub fn $name(name: &str, val: Term) -> ExSeries {
+        pub fn $name(name: &str, val: Term) -> NifResult<ExSeries> {
             let nan = atoms::nan();
             let infinity = atoms::infinity();
             let neg_infinity = atoms::neg_infinity();
 
-            ExSeries::new(Series::new(
-                name,
-                val.decode::<ListIterator>()
-                    .unwrap()
-                    .map(|item| match item.get_type() {
-                        TermType::Number => Some(item.decode::<$type>().unwrap()),
-                        TermType::Atom => {
-                            if nan.eq(&item) {
-                                Some($module::NAN)
-                            } else if infinity.eq(&item) {
-                                Some($module::INFINITY)
-                            } else if neg_infinity.eq(&item) {
-                                Some($module::NEG_INFINITY)
-                            } else {
-                                None
-                            }
-                        }
-                        term_type => panic!("from_list/2 not implemented for {term_type:?}"),
-                    })
-                    .collect::<Vec<Option<$type>>>(),
-            ))
+            let values: NifResult<Vec<Option<$type>>> = val
+                .decode::<ListIterator>()?
+                .map(|item| match item.get_type() {
+                    TermType::Number => item.decode::<Option<$type>>(),
+                    TermType::Atom => Ok(if nan.eq(&item) {
+                        Some($module::NAN)
+                    } else if infinity.eq(&item) {
+                        Some($module::INFINITY)
+                    } else if neg_infinity.eq(&item) {
+                        Some($module::NEG_INFINITY)
+                    } else {
+                        None
+                    }),
+                    term_type => {
+                        let message = format!("from_list/2 not implemented for {term_type:?}");
+                        Err(Error::RaiseTerm(Box::new(message)))
```

Comment:
It doesn't matter much honestly. As long as it comes out properly when we show it to the user.

Reply:
There are other places in `series.rs` file where `panic!` is being used - won't it crash VM ? Is it  because that validations are  done in Elixir these panic clause won't ever trigger ?

Reply:
Thank you

Reply:
@philss one last question, does panic! behave differently in dev vs release builds?

Reply:
@lkarthee no, it shouldn´t be different. I guess the problem is related to the lib that catches the panics.

> libunwind: stepWithCompactEncoding - invalid compact unwind encoding

I'm not sure, but looks like this is some encoding problem there.